1. Why are functions advantageous to have in your programs?
Ans1-Because we can reuse the code ,without doing complete code we can call the function anytime.

2. When does the code in a function run: when it's specified or when it's called?
Ans2-the code in a function runs, when it's called.

3. What statement creates a function?
Ans3- def ABC() : creates the function with ABC as the function name.

4. What is the difference between a function and a function call?
Ans4- In function we can pass the data, parameters,etc but it will not executed until it calls,
When a program calls a function, the program control is transferred to the called function

5.How many global scopes are there in a Python program? How many local scopes?
Ans5-There is one global scope, and a local scope is created whenever a function is called.

6.What happens to variables in a local scope when the function call returns?
Ans6-

7. What is the concept of a return value? Is it possible to have a return value in an expression?
Ans7-

8. If a function does not have a return statement, what is the return value of a call to that function?
Ans8-

9. How do you make a function variable refer to the global variable?
Ans9-

10.What is the data type of None?
Ans10-None is a data type of its NoneType

11. What does the sentence import areallyourpetsnamederic do?
Ans11-


12. If you had a bacon() feature in a spam module, what would you call it after importing spam?
Ans12-spam. bacon()


13.What can you do to save a programme from crashing if it encounters an error?
Ans13-When it throws error on the programme then it directly move to the except block,in which it skip the code in betweeen.Try running the program and it should throw an error message instead of crashing the program.


14.What is the purpose of the try clause? What is the purpose of the except clause?
Ans14-If any error occurs on the code then it throws error and the code will not be executed.So we use try and except in which The try block lets you test a block of code for errors. The except block lets you handle the error